src/ProblemBrowser/ProblemBrowserSearchPage.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState } from 'react';
+import { Tabs, Tab, Container } from 'react-bootstrap';
+import { useHistory, useRouteMatch } from 'react-router-dom';
+import { useQuery } from '../Hooks/UseQuery';
+import { ProblemBrowserOpenProblemLibraryForm } from './ProblemBrowserOpenProblemLibraryForm';
+import { ProblemBrowserCourseForm } from './ProblemBrowserCourseForm';
+import { ProblemBrowserPrivateForm } from './ProblemBrowserPrivateForm';
+
+interface ProblemBrowserSearchPageProps {
+
+}
+
+enum ProblemBrowserSearchPageTabs {
+    OPEN_PROBLEM_LIBRARY = 'Open Problem Library',
+    COURSES = 'Courses',
+    PRIVATE_CONTENT = 'Private Content'
+}
+
+export const ProblemBrowserSearchPage: React.FC<ProblemBrowserSearchPageProps> = () => {
+    const queryParams = useQuery();
+    const history = useHistory();
+    const { url } = useRouteMatch();
+    const [activeTab, setActiveTab] = useState<ProblemBrowserSearchPageTabs>(ProblemBrowserSearchPageTabs.OPEN_PROBLEM_LIBRARY);
+    const tab = queryParams.get('tab') as ProblemBrowserSearchPageTabs | null;
+
+    // TODO: Back navigation with this approach seems slow. Is there a faster way to detect the url change?
+    useEffect(()=>{
+        setActiveTab(tab ?? ProblemBrowserSearchPageTabs.OPEN_PROBLEM_LIBRARY);
```

Comment:
Do you need this if you're calling `setActiveTab` when you're updating the history? You might be able to just use `tab` as a default for the `useState` on line 23 rather than updating it via a useEffect.

Reply:
I don't know I think I copied what you did 🤣 maybe a future fix, it already has TODO on it 😆